src/plugins/data/public/search/errors/painless_error.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { get } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { EuiButton, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { ApplicationStart } from 'kibana/public';
+import { KbnError } from '../../../../kibana_utils/common';
+import { IEsSearchRequest } from '..';
+
+export class PainlessError extends KbnError {
+  constructor(err: Error, request: IEsSearchRequest) {
+    const rootCause = get(err, 'body.attributes.error.root_cause');
+    const [{ script }] = rootCause;
+
+    super(
+      i18n.translate('data.painlessError.painlessScriptedFieldErrorMessage', {
+        defaultMessage: "Error with Painless scripted field '{script}'.",
+        values: { script },
+      })
+    );
+  }
+
+  public getErrorMessage(application: ApplicationStart) {
+    function onClick() {
+      application.navigateToApp('management', {
+        path: `/kibana/indexPatterns`,
+      });
+    }
+
+    return (
+      <>
+        {this.message}
+        <EuiSpacer size="s" />
+        <div className="eui-textRight">
+          <EuiButton color="danger" onClick={onClick} size="s">
+            <FormattedMessage id="data.painlessError.buttonTxt" defaultMessage="Edit script" />
+          </EuiButton>
+        </div>
+      </>
+    );
+  }
```

Comment:
Could be add more context about the error to the user? this would be very helpful, at least there should be a possibility to view the raw error, since it contains more info about what was going wrong

Reply:
This is the format I created - I think it's good enough as the actual error just shows you the stack trace to the search interceptor where the error was thrown.

![image](https://user-images.githubusercontent.com/3016806/93879715-e8ff2600-fce4-11ea-8559-e6d953ab1d3c.png)